### ISSUE TITLE
"Failed to connect" error

### ISSUE THREAD

**Issue by manwithpowers** [rrooggiieerr/homeassistant-benqprojector#22]
I'm trying to connect my BenQ W700 projector via this integration and keep getting a "Failed to connect" error.
In the logs all I see is `Failed to get projector command prompt, is your projector properly connected?` (Looks like this is the source of the error `Source: custom_components/benqprojector/config_flow.py:157`)

I have tested using minicom and I'm able to communicate with the projector (on, off, modelname), I've made sure the baud rate is the same on the projector as I'm entering here, and I've tested with the projector both on and off already.

The one thing that sticks out to me is that it's timing out very quickly - almost immediately after clicking "Submit" when adding the projector, I get the above error.

Let me know if there's any other info I can provide to help troubleshoot this and thanks in advance for any help!
<img width="584" alt="Screenshot 2024-10-27 at 18 08 29" src="https://github.com/user-attachments/assets/cb68cd4a-136f-4e98-bbd8-6fb46bdb69b0">

**Comment by rrooggiieerr** [comment]
Can you show me a screenshot of the minicom session?

**Comment by manwithpowers** [comment]
Sure thing:
<img width="487" alt="Screenshot 2024-10-27 at 18 49 55" src="https://github.com/user-attachments/assets/a662fc5f-7870-4267-ba57-77c4cb0676c9">

I also tested using benqprojector.py directly and got this
<img width="662" alt="Screenshot 2024-10-27 at 18 54 36" src="https://github.com/user-attachments/assets/6db0510e-3101-443a-8cdf-6f055824d2f0">

**Comment by rrooggiieerr** [comment]
Ok, minicom output looks exactly like it should

Can you add the `--debug` argument to the `benqprojector` CLI?

**Comment by rrooggiieerr** [comment]
Sorry, getting back to the minicom output, it actually doesn't look like it should. I'm expecting to see `*POW=ON#` and `*MODELNAME=W700#` returned on the commands

**Comment by manwithpowers** [comment]
Here's the debug output:
<img width="1053" alt="Screenshot 2024-10-27 at 19 11 42" src="https://github.com/user-attachments/assets/77b50616-5e53-4add-93e6-a5316f852e3c">

Minicom does actually show `*POW=ON#` response but for whatever reason in my terminal that gets overwritten when I type a new line (here it is on it's own):
<img width="176" alt="Screenshot 2024-10-27 at 19 12 16" src="https://github.com/user-attachments/assets/d78faa52-1407-4b1e-8ffc-5975f5ec5a6a">
Response to `*modelname=?#` is however just `W700`

Thanks for the quick replies!

**Comment by rrooggiieerr** [comment]
Is the warning message consistent when you run the `benqprojector` CLI command a couple of times?

**Comment by manwithpowers** [comment]
It's now slightly different than that first run using `--debug` but is now consistent:
<img width="1067" alt="Screenshot 2024-10-27 at 19 28 51" src="https://github.com/user-attachments/assets/63be063f-2abd-4569-9fb7-f6219cf575f5">

**Comment by rrooggiieerr** [comment]
Are you sure the baudrate is correct? You can find the baudrate in the on screen menu of your projector

**Comment by manwithpowers** [comment]
Yes, it's definitely correct
![IMG20241027193501](https://github.com/user-attachments/assets/b8598ab9-008e-40f5-9347-26caffc5b4d0)

**Comment by rrooggiieerr** [comment]
Can you set it higher and see if that helps? Maybe there is indeed a timeout in the code that exceeds with low baud rates. I run my serial connection on 115200 baud

**Comment by manwithpowers** [comment]
Yeah, 115200 is what I had mine set to originally, as well. I dropped it down based on some feedback I got elsewhere when I was just testing the cable. Tested again at 115200 and at 57600 a few times and unfortunately still no dice:
<img width="1106" alt="Screenshot 2024-10-27 at 20 03 16" src="https://github.com/user-attachments/assets/b219610c-0d07-4e2a-b670-48480771ce0c">

**Comment by rrooggiieerr** [comment]
Could you try from a different machine?

**Comment by manwithpowers** [comment]
Using `benqprojector` CLI? Yeah, I can give that a shot from my laptop tomorrow. I tested the cable and projector connection using my laptop & minicom first so I know that part works. Thanks for all the help!

**Comment by manwithpowers** [comment]
Unfortunately, `benqprojector` CLI also doesn't work from my laptop (tried a few different commands - the first two are with the projector off, the rest the projector was on). As mentioned earlier, I am able to connect to the projector using `minicom` or `screen` on this machine.
<img width="893" alt="Screenshot 2024-10-28 at 12 51 29" src="https://github.com/user-attachments/assets/ce712955-dcea-40da-98fa-15e1daebf02c">

**Comment by rrooggiieerr** [comment]
Ok, I think the problem here is the `\x00` after the prompt which I haven't seen before in other BenQ projectors. I need to think about this.
